lib/matplotlib/pyplot.py

```diff
@@ -1552,19 +1552,19 @@ def yscale(scale, **kwargs):
     gca().set_yscale(scale, **kwargs)
 
 
-def xticks(*args, **kwargs):
+def xticks(ticks=None, labels=None, **kwargs):
     """
     Get or set the current tick locations and labels of the x-axis.
 
     Call signatures::
 
         locs, labels = xticks()           # Get locations and labels
 
-        xticks(locs, [labels], **kwargs)  # Set locations and labels
+        xticks(ticks, [labels], **kwargs)  # Set locations and labels
 
     Parameters
     ----------
-    locs : array_like
+    ticks : array_like
```

Comment:
Not really happy with the name `ticks`. `locs` or better `locations` would be more descriptive. However, `ticks` is used in `Axes.set_x/yticks`.

Personally, I'd favor better naming over consistency here. Also, I would rename the `Axes.set_x/yticks` parameter name with a deprecation cycle and compatibilility code for a transition phase. Anyway I believe that this parameter is mostly used positional, so it will only affect very little code. I think the better naming in the docs and function signature is worth such a change. That said, I would also understand if other devs are more defensive about it.

Reply:
@anntzer - do you have a problem with naming this argument `locations`?

If not, I'll rename it but leave the `Axes.set_x/yticks` renaming for another PR.

Reply:
Having a signature `set_xticks(ticks)` does say nothing about the semantics of the parameter (would be equally bad as `set_xticks(value)`). Do I have to pass positions or labels or a list of Tick instances?

In contrast `set_xticks(positions)` conveys meaning.

Reply:
Fine, but I'm not in favour of making the two calls inconsistent, and we can have that discussion when a new PR that proposes changing them both.

Reply:
Fine by me.

Reply:
OK; I'll leave this named `ticks` and changes can go in a later PR.

Reply:
I actually agree that `locations` is better, but that should be done in both places at the same time.